docs/source/docs/configuration.md

```diff
@@ -2,19 +2,592 @@
+
+#### Regularization
+
+The regularization section configures the losses for the DDF. To instantiate this part
+of the loss, pass "regularization" into the config file as a field.
+
+- `weight`: float type, weight of the regularization loss.
+- `name`: string type, the type of deformation energy to compute. Options include
+  "bending", "gradient"
+
+If the `gradient` loss is used, another optional argument can be passed at the same
+indent level: - `l1`: bool, optional, default is false. Indicates whether to calculate
+the L1 (true) or L2 (false) loss by modifying the bool argument.
+
+EG.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    regularization:
+      weight: 0.5 # weight of regularization loss
+      name: "bending" # options include "bending", "gradient"
+```
+
+or
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    regularization:
+      weight: 0.5 # weight of regularization loss
+      name: "gradient" # options include "bending", "gradient"
+      l1: false
+```
+
+#### Multiple Losses
+
+Add multiple losses by putting all the fields in the same file.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    image:
+      name: "gmi"
+      weight: 1.0
+    label:
+      weight: 1.0
+      name: "dice"
+```
+
+### Optimizer - required
+
+The optimizer can be defined by using a `name` and then passing optimizer specific
+arguments with the same name. All optimizers can use the `learning_rate` argument.
+
+- `name`: string type, is used to define the optimizer during training. One of "adam",
+  "sgd", "rms". There must be another additional field with the same name.
+
+- `adam`: If adam is passed into `name`, the `adam` field must be passed. The dictionary
+  can be empty, which initalises a default
+  [Keras Adam optimizer](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Adam).
+  Alternatively, fields with names equivalent to those specified in the optimizer
+  documentation can be used.
+- `sgd`: If sgd is passed into `name`, the `sgd` field must be passed. The dictionary
+  can be empty, which initalises a default
+  [Keras SGD optimizer](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/SGD).
+  Alternatively, fields with names equivalent to those specified in the optimizer
+  documentation can be used instead.
```

Comment:
"can be used" - > "can be configured by adding additional field with the same name" here?

Reply:
with the following examples, it is clear maybe...